packages/loader/container-loader/src/container.ts

```diff
@@ -1549,6 +1549,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
         deltaManager.on("disconnect", (reason: string) => {
             this.manualReconnectInProgress = false;
+            // Register submitOp event again as we only want to listen for first op from this client.
+            this.registerSubmitOpEvent();
             this.connectionStateHandler.receivedDisconnectEvent(reason);
```

Comment:
is it possible to add test for the scenario we missed

Reply:
also, i think we can do this another way that is potentially simpler. if our client id is in the quorum, we need to wait for leave, otherwise we do not, as we can't send ops if we are not in the quorum

Reply:
I had tests for multiple disconnects but the tests are unit tests for the connection handler class, so the firing of event/calling function was was done manually.

Reply:
Actually this is for case where the client did connected and is in the quorum, but this client did not send any ops by itself, so we don't need to wait for its leave op. Only the client which send the ops should we awaited for leave op.

Reply:
I'd much rather keep this simpler, and wait for clients to leave if they are in the quorum, rather than doing any fancy op tracking

Reply:
given we have read and write mode clients should really only be in the quorum if they send an op anyway